### PR TITLE
Convert gitlab_group_members's `members` field from TypeList to TypeSet

### PR DESCRIPTION
**Pull request by pdecat**
Also, remove gitlab_group_members redundant top level fields: group_owner_id, access_level, expires_at

### CODE REVIEW

File: scripts/start-gitlab.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ then
         exit 1
     fi
 else
-    img=gitlab/gitlab-ce
+    img=gitlab/gitlab-ce:13.0.6-ce.0
```
Comment (pdecat) [inline scripts/start-gitlab.sh]: Work-around for https://github.com/terraform-providers/terraform-provider-gitlab/issues/338#issuecomment-647694847

Comment (pdecat) [reply scripts/start-gitlab.sh]: The deploy key issue was resolved by Gitlab 13.1.1 so I reverted this change but this now JIRA service tests fail:

```sh
# MAKE_TARGET=testacc scripts/start-gitlab.sh
Starting gitlab container...
dfd4ea69b610401cf74ff8a1f5d2803d0ee460ef2bc2a4679925db15b5746869
Waiting for gitlab to be ready ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... .
Creating access token



--------------------------------------------------------------------------------
 GitLab:       13.1.1 (bdb9883705a) FOSS
 GitLab Shell: 13.3.0
 PostgreSQL:   11.7
--------------------------------------------------------------------------------
Loading production environment (Rails 6.0.3.1)
Switch to inspect mode.
terraform_token = PersonalAccessToken.create(user_id: 1, scopes: [:api, :read_user], name: :terraform);terraform_token.set_token('ACCTEST');terraform_token.save!;
true

# make testacc TEST=./gitlab TESTARGS='-run TestAccGitlabServiceJira_basic -count 1'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test -v ./gitlab -run TestAccGitlabServiceJira_basic -count 1 -timeout 40m
=== RUN   TestAccGitlabServiceJira_basic
    TestAccGitlabServiceJira_basic: testing.go:684: Step 1 error: Check failed: Check 2/8 error: gitlab_service_jira.jira: Attribute 'url' expected "https://testurl.com", got "https://test.com"
--- FAIL: TestAccGitlabServiceJira_basic (10.67s)
FAIL
FAIL    github.com/terraform-providers/terraform-provider-gitlab/gitlab 10.684s
FAIL
make: *** [GNUmakefile:23: testacc] Error 1
```

https://github.com/claranet/terraform-provider-gitlab/runs/810546410

Comment (pdecat) [reply scripts/start-gitlab.sh]: The pinning was reinstated to work this around for now.

File: gitlab/resource_gitlab_group_members.go [hunk 2]
```diff
@@ -35,69 +48,49 @@ func resourceGitlabGroupMembers() *schema.Resource {
...
 
+func hashGroupMembersMembers() schema.SchemaSetFunc {
+	return schema.HashResource(schemaGroupMembersMembers())
+}
+
 func resourceGitlabGroupMembersCreate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*gitlab.Client)
 
 	groupID := d.Get("group_id").(string)
-	groupOwnerID := d.Get("group_owner_id").(int)
```
Comment (maxenced) [inline gitlab/resource_gitlab_group_members.go:88]: Looks like a  breaking change, should this be documented / have some backward compatibility ?

File: scripts/start-gitlab.sh (hunk 1 above)
Comment (pdecat) [reply scripts/start-gitlab.sh]: Removed the pinning as the issue mysteriously no longer occurs: https://github.com/terraform-providers/terraform-provider-gitlab/issues/353

File: gitlab/resource_gitlab_group_members.go (hunk 2 above)
Comment (maxenced) [reply gitlab/resource_gitlab_group_members.go:88]: @pdecat any opinion ?

**Comment by pdecat** [comment]
@maxenced @Shr3ps gentle ping :stuck_out_tongue:

**Review by Shr3ps — APPROVED** [review]

**Review by maxenced — APPROVED** [review]

File: gitlab/resource_gitlab_group_members.go (hunk 2 above)
Comment (pdecat) [reply gitlab/resource_gitlab_group_members.go:88]: It is indeed a breaking change and will be listed as such in our fork's CHANGELOG.
However, it never made it upstream so I believe it's okay. I'd prefer not to manage backward compatibility for this custom resource until it's upstream.

**Pull request merged**